### ISSUE TITLE
Fix EventMesh upgrade test on GKE

### ISSUE THREAD

**Issue by marcobebway** [kyma-project/kyma#8280]
**Description**

The `post-master-kyma-gke-upgrade` prow job is failing because of the `EventMeshUpgradeTest` failure.

**Job**

https://status.build.kyma-project.io/view/gcs/kyma-prow-logs/logs/post-master-kyma-gke-upgrade/1255065939431395329.

**Logs**
```json
{
   "level":"error",
   "log":{
      "message":"[TestResources: EventMeshUpgradeTest] End with error [start time: Tue, 28 Apr 2020 10:18:30 UTC, duration: 52.121093655s]: waiting for application failed: All attempts fail:\n#10: knative-service eventmesh-upgrade for HTTPSource not ready, ksvc: \u0026{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:eventmesh-upgrade GenerateName: Namespace:kyma-integration SelfLink:/apis/serving.knative.dev/v1alpha1/namespaces/kyma-integration/services/eventmesh-upgrade UID:315c674c-ddbf-4be5-9e32-8b4b8a4e95a7 ResourceVersion:19483 Generation:6 CreationTimestamp:2020-04-28 09:49:05 +0000 UTC DeletionTimestamp:\u003cnil\u003e DeletionGracePeriodSeconds:\u003cnil\u003e Labels:map[application-name:eventmesh-upgrade serving.knative.dev/visibility:cluster-local] Annotations:map[serving.knative.dev/creator:system:serviceaccount:kyma-system:event-sources-controller-manager serving.knative.dev/lastModifier:system:serviceaccount:kyma-system:event-sources-controller-manager] OwnerReferences:[{APIVersion:sources.kyma-project.io/v1alpha1 Kind:HTTPSource Name:eventmesh-upgrade UID:14a90445-a9cb-4b3d-ad57-39c05866fdbb Controller:0xc00086029e BlockOwnerDeletion:0xc00086029d}] Initializers:nil Finalizers:[] ClusterName: ManagedFields:[]} Spec:{DeprecatedGeneration:0 DeprecatedRunLatest:\u003cnil\u003e DeprecatedPinned:\u003cnil\u003e DeprecatedManual:\u003cnil\u003e DeprecatedRelease:\u003cnil\u003e ConfigurationSpec:{DeprecatedGeneration:0 DeprecatedBuild:nil DeprecatedRevisionTemplate:\u003cnil\u003e Template:\u0026ObjectMeta{Name:,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:\u003cnil\u003e,DeletionGracePeriodSeconds:nil,Labels:map[string]string{kyma-project.io/dashboard: event-mesh,},Annotations:map[string]string{autoscaling.knative.dev/minScale: 1,},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,ManagedFields:[],}} RouteSpec:{DeprecatedGeneration:0 Traffic:[{DeprecatedName: TrafficTarget:{Tag: RevisionName: ConfigurationName: LatestRevision:0xc000860710 Percent:0xc000860718 URL:}}]}} Status:{Status:{ObservedGeneration:6 Conditions:[{Type:ConfigurationsReady Status:Unknown Severity: LastTransitionTime:{Inner:2020-04-28 10:04:37 +0000 UTC} Reason: Message:} {Type:Ready Status:False Severity: LastTransitionTime:{Inner:2020-04-28 10:04:38 +0000 UTC} Reason:RevisionMissing Message:Revision \"eventmesh-upgrade-czhrc\" referenced in traffic not found.} {Type:RoutesReady Status:False Severity: LastTransitionTime:{Inner:2020-04-28 10:04:38 +0000 UTC} Reason:RevisionMissing Message:Revision \"eventmesh-upgrade-czhrc\" referenced in traffic not found.}]} RouteStatusFields:{URL:http://eventmesh-upgrade.kyma-integration.svc.cluster.local DeprecatedDomain: DeprecatedDomainInternal: Address:0xc000a4a1e0 Traffic:[{DeprecatedName: TrafficTarget:{Tag: RevisionName:eventmesh-upgrade-czhrc ConfigurationName: LatestRevision:0xc00086080b Percent:0xc000860ab0 URL:}}]} ConfigurationStatusFields:{LatestReadyRevisionName:eventmesh-upgrade-czhrc LatestCreatedRevisionName:eventmesh-upgrade-z4jtc}}}, application: \u0026{TypeMeta:{Kind:Application APIVersion:applicationconnector.kyma-project.io/v1alpha1} ObjectMeta:{Name:eventmesh-upgrade GenerateName: Namespace: SelfLink:/apis/applicationconnector.kyma-project.io/v1alpha1/applications/eventmesh-upgrade UID:c955ac8e-d3ef-473b-afaf-98a3c167108f ResourceVersion:19869 Generation:3 CreationTimestamp:2020-04-28 09:49:02 +0000 UTC DeletionTimestamp:\u003cnil\u003e DeletionGracePeriodSeconds:\u003cnil\u003e Labels:map[] Annotations:map[] OwnerReferences:[] Initializers:nil Finalizers:[finalizer.applicationconnector.kyma-project.io] ClusterName: ManagedFields:[]} Spec:{Description: SkipInstallation:false Services:[{ID:eventmesh-upgrade Identifier: Name:eventmesh-upgrade DisplayName:Some Events Description:Application Service Class with Events Labels:map[connected-app:app-name] LongDescription: ProviderDisplayName:provider Tags:[] Entries:[{Type:Events GatewayUrl: AccessLabel: TargetUrl: SpecificationUrl: ApiType: Credentials:{Type: SecretName: AuthenticationUrl: CSRFInfo:\u003cnil\u003e} RequestParametersSecretName:}]}] AccessLabel:eventmesh-upgrade Labels:map[] Tenant: Group: CompassMetadata:\u003cnil\u003e} Status:{InstallationStatus:{Status:DEPLOYED Description:Upgrade complete}}}",
      "taskID":"a7d64202-9ed4-4262-a3b4-d818ec7cf8b2",
      "time":"2020-04-28T10:19:22.847Z"
   }
}
```

```
knative-service eventmesh-upgrade for HTTPSource not ready
```

**AC**

- The `EventMeshUpgradeTest` is fixed and the `post-master-kyma-gke-upgrade` is green.

**Comment by sayanh** [comment]
Following is happening:
- As event-sources-controller-manager gets upgraded, the new pod updates the `ksvc` for the `HTTPSource` as well.
- As the ksvc gets modified, this leads to the creation of a new `Revision`.
- The serving-controller now fails to pick up the new `Revision` and marks the ksvc to be `Not ready` as it still looks for the old `Revision`, which is gone.
- This is a known bug in `0.12` and was fixed in 0.13+ knative-serving.
- PR: https://github.com/knative/serving/pull/6592
- SHA: https://github.com/knative/serving/commit/627315be8d5e4789f1d3aabf83d628989e3b7351)
- This issue can be fixed with knative-serving upgrade, which will be taken care in this [issue](https://github.com/kyma-project/kyma/issues/8351).

**Comment by marcobebway** [comment]
LGTM.
